driver/src/main/com/mongodb/DB.java

```diff
@@ -617,4 +619,73 @@ private BsonDocument wrap(final DBObject document, final DBEncoder encoder) {
             return new BsonDocumentWrapper<DBObject>(document, new DBEncoderAdapter(encoder));
         }
     }
+
+    /**
+     * Determines the read preference that should be used for the given command.
+     *
+     * @param command             the {@link DBObject} representing the command
+     * @param requestedPreference the preference requested by the client.
+     * @return the read preference to use for the given command.  It will never return {@code null}.
+     * @see com.mongodb.ReadPreference
+     */
+    ReadPreference getCommandReadPreference(final DBObject command, final ReadPreference requestedPreference) {
+        if (mongo.getReplicaSetStatus() == null) {
+            if (requestedPreference == null) {
+                return ReadPreference.primary();
+            } else {
+                return requestedPreference;
+            }
+        }
+
+        String comString = command.keySet().iterator().next().toLowerCase();
+
+        if (comString.equals("getnonce") || comString.equals("authenticate")) {
```

Comment:
This was here in 2.x to support DB.authenticate, which no longer exists.  I think we can remove.